src/main/java/org/opentripplanner/routing/algorithm/transferoptimization/package.md

```diff
@@ -32,30 +32,36 @@ two trips in a given path.
        having 30 seconds for one and 7m30s for the other.  
     3. For a transfer between two trips we want to find the right balance, between
        extra-transfer-time and the path generalized-cost(like the cost of walking and riding a
-       buss). 
+       bus). 
  
-## Not supported
+## Not Supported
  - Using GTFS transfers.txt it is possible to set the `min_transfer_time` with `transfer_type = 2`. 
  - The NeTEx Interchange MaximumWaitTime is ignored.
 
-## Process
+## Implementation
 
 Finding the best transfers is done in 2 separate steps in OTP:
 
- 1. As part of the routing. The routing engine (Raptor) has a generalized-cost and 
-    number-of-transfers as criteria during the routing. Some of the goals above are implemented 
-    using the generalized-cost.
- 2. Post processing paths. Paths from the routing search are optimized with respect to where 
-    transfers happen for each pair of transit rides(trips). This step do NOT compare two paths 
-    returned by the router(Raptor), but instead find all possible transfers for a given path, and 
-    the different alternatives to transfer between each pair of trip within the path. This is an 
-    outline of the process:
+ 1. As part of the routing. The routing engine (Raptor) applies generalized-cost and 
+    number-of-transfers as criteria during the routing. In addition, Raptor supports overriding 
+    regular transfers with guaranteed transfers. 
+    1. Goal 1 is achieved by having `number-of-transfers` as a Raptor criterion.
+    2. Goal 2 is achieved by giving some stops a lower _visiting-cost_ according to the 
+       `StopTransferPriority`.
+    3. Goal 3 and 6 is achieved by allowing guaranteed transfers to override regular transfers. An 
+       optional `RaptorGuaranteedTransferProvider` is injected into Raptor witch Raptor calls to 
+       get guaranteed transfers. This service is also responsible for rejecting a transfer. 
+ 2. Goal 4, 5, and 7 are achieved by post-processing paths. Paths from the routing search are revised,
+    optimizing the stop at which transfers happen between each pair of transit rides (trips). This
+    step does NOT compare different paths returned by the router (Raptor), but instead finds all possible
+    transfer locations for a given path, and the different alternatives to transfer between each pair of 
+    trips within the path. This is an outline of the process:
     1. For each path find all possible permutations of transfers.
     2. Filter paths based on priority including `StaySeated=100`, `Guaranteed=10`, `Preferred=2`, 
```

Comment:
This item 2 is a bit confusing as it shows costs (priorities) for different kinds of transfers, but says it's filtering paths based on priority. When first reading this it sounds like different paths (different unique sequences of trips) are being filtered (removed), but it seems more likely here that specific transfer points for each path are being ordered by priority. Perhaps the word "path" is being used here to mean a sequence of trips _with_ fully determined transfer points; in other places the term seems to mean a sequence of trips _without_ specific determined transfer points.